src/jdk.crypto.mscapi/windows/classes/sun/security/mscapi/CKeyStore.java

```diff
@@ -746,9 +755,13 @@ public void engineStore(OutputStream stream, char[] password)
      */
     public void engineLoad(InputStream stream, char[] password)
             throws IOException, NoSuchAlgorithmException, CertificateException {
-        if (stream != null && !keyStoreCompatibilityMode) {
+        if (stream != null && !getRequiresKeyStoreFromFile() && !keyStoreCompatibilityMode) {
```

Comment:
Not sure of this logic.  It's saying that a warning message will get through if the stream is not null AND it's not loading a key from a file AND its not running in compatibility mode.  There seems some states in the state machine not covered?